packages/platform-android/src/commands/runAndroid/index.ts

```diff
@@ -115,15 +109,18 @@ function getPackageNameWithSuffix(
 }
 
 // Builds the app and runs it on a connected emulator / device.
-function buildAndRun(args: Flags) {
-  process.chdir(path.join(args.root, 'android'));
-  const cmd = process.platform.startsWith('win') ? 'gradlew.bat' : './gradlew';
+function buildAndRun(args: Flags, config: Config) {
+  const androidConfig = config.project.android;
+  if (!androidConfig) {
+    return;
+  }
+  process.chdir(androidConfig.sourceDir);
```

Comment:
While we are at it, I am wondering if we should remove this and prefer setting `cwd` explicitly while running a command or fixing paths in other functions? Would likely help with testing (similar what we did with PackageManager).

Reply:
If not, maybe we can make this a first-line, on line 42? That way we can keep a great balance of making it explicit and working inside "android" folder if it makes things easier.